src/main/java/ch/njol/skript/entity/BoatChestData.java

```diff
@@ -24,28 +24,41 @@
 import ch.njol.skript.lang.Literal;
 import ch.njol.skript.lang.SkriptParser;
 import org.bukkit.Material;
-import org.bukkit.TreeSpecies;
+import org.bukkit.entity.Boat;
 import org.bukkit.entity.ChestBoat;
 import org.bukkit.inventory.ItemStack;
-import org.eclipse.jdt.annotation.Nullable;
+import org.jetbrains.annotations.Nullable;
 
+import java.util.Locale;
 import java.util.Random;
 
 public class BoatChestData extends EntityData<ChestBoat> {
 
+	private static final Boat.Type[] types = Boat.Type.values();
+
 	static {
+		// This ensures all boats are registered
+		// As well as in the correct order via 'ordinal'
+		String[] patterns = new String[types.length + 2];
+		patterns[0] = "chest boat";
+		patterns[1] = "any chest boat";
+		for (Boat.Type boat : types) {
+			String boatName = boat.toString().replace("_", " ").toLowerCase(Locale.ENGLISH) + " chest boat";
+			if (boatName.startsWith("bamboo"))
+				boatName = "bamboo chest raft";
+			patterns[boat.ordinal() + 2] = boatName;
```

Comment:
this is a bit awkward, could we just check if the type is bamboo instead of doing all the string stuff?